Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -530,7 +530,11 @@ public class AttributeRemover: SyntaxRewriter {
 
   public override func visit(_ node: AttributeListSyntax) -> AttributeListSyntax {
     var filteredAttributes: [AttributeListSyntax.Element] = []
-    for case .attribute(let attribute) in node {
+    for attribute in node {
+      guard case .attribute(let attribute) = attribute else {
+        filteredAttributes.append(attribute)
```

Comment:
What if `#if ... #endif` contained attributes matching `self.predicate(attribute)`? Or it never happens because MacroSystem doesn't support `#if` attributes?

Reply:
My thought is that `MacroSystem` doesn’t know how to evaluate `#if` conditions, so just keeping them verbatim is the best we can do…

Reply:
I see `MacroApplication.macroAttributes(attachedTo:)` doesn't look into `#if` at all at this point. So yeah, this is fine.